File: Spa-Salon-FrontEnd/src/types/customer.ts
import type { AppointmentStatus } from './common';

export interface Customer {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  hasUserAccount: boolean;
}

export interface CustomerAppointment {
  id: string;
  appointmentDateTime: string;
  status: AppointmentStatus;
  totalPrice: number;
  services: AppointmentServiceSummary[];
}

export interface AppointmentServiceSummary {
  serviceName: string;
  employeeName: string;
  price: number;
}

export interface CustomerDetail extends Customer {
  dateOfBirth?: string;
  userId?: string;
  appointments: CustomerAppointment[];
}

export interface CreateCustomer {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  phone?: string;
  dateOfBirth?: string;
}

export interface UpdateCustomer {
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  dateOfBirth?: string;
}

File: Spa-Salon-BackEnd/Domain/Entities/Customer.cs
namespace Spa_Salon_BackEnd.Domain.Entities;

public class Customer
{
    public Guid Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public DateTime? DateOfBirth { get; set; }
    public Guid? UserId { get; set; }

    public User? User { get; set; }
    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
}
